plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/revisited/modules/SourceModule.java

```diff
@@ -694,10 +696,38 @@ public TokensList getCompletionsForBase(ICompletionState initialState, ICodeComp
             state = initialState.getCopy();
             state.setActivationToken(s);
             return manager.getCompletionsForModule(this, state);
+        } else if (classDef.bases[baseIndex] instanceof Subscript) {
+            TokensList tokens = new TokensList();
+            Subscript subscript = (Subscript) classDef.bases[baseIndex];
+            tokens.addAll(getCompletionsForValueAsName(manager, initialState, subscript.value));
+            if (subscript.slice instanceof Index) {
+                Index index = (Index) subscript.slice;
+                tokens.addAll(getCompletionsForValueAsName(manager, initialState, index.value));
+            }
+            if (tokens.size() > 0) {
+                return tokens;
+            }
         }
         return null;
     }
 
+    private TokensList getCompletionsForValueAsName(ICodeCompletionASTManager manager, ICompletionState state,
+            exprType value) throws CompletionRecursionException {
+        if (value instanceof Name) {
+            Name name = (Name) value;
+            if (name.id != null && !name.id.isEmpty()) {
+                ICompletionState copiedState = state.getCopy();
+                state.checkMemory(this, name.id);
+                copiedState.setActivationToken(name.id);
+                TokensList subscriptValueToks = manager.getCompletionsForModule(this, copiedState);
+                if (subscriptValueToks != null && subscriptValueToks.size() > 0) {
```

Comment:
I think you don't need to check the size here (if it's empty, it's ok to return it at this point, only the null check is needed).